specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-03-01-preview/apimgateways.json

```diff
@@ -1318,6 +1318,154 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/gateways/{gatewayId}/invalidateDebugCredentials": {
+      "post": {
+        "tags": [
+          "GatewayInvalidateDebugCredentials"
+        ],
+        "operationId": "Gateway_InvalidateDebugCredentials",
+        "description": "Action is invalidating all debug credentials issued for gateway.",
+        "x-ms-examples": {
+          "ApiManagementGatewayInvalidateDebugCredentials": {
+            "$ref": "./examples/ApiManagementGatewayInvalidateDebugCredentials.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/GatewayIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "All debug credentials for gateway are now invalidated."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/gateways/{gatewayId}/getDebugCredentials": {
```

Comment:
>get [](http://example.com/codeflow?start=149&length=3)

We recommend using the verbs other than get in the post actions. This could be renamed to listDebugCredentials

Reply:
`list` does not work with the thing which we want to accomplish here. The endpoint is for receiving debug credential. That's why `get` prefix is there.

What about `createDebugCredential`? Is it acceptable?

For getTrace, what about `showTrace` ? Is it acceptable?

Reply:
@sjanamma We do not see a usages of that API from ARM templates. Is it still required to have both of that actions `list*` prefixed?

Reply:
`createDebugCredentials` and `showTrace` are acceptable names. Only get is not correct.

Reply:
After discussion with @VitaliyKurokhtin changed to `list` prefix (`listDebugCredentials`, `listTrace`)